src/ppl/src/IOPlacer.cpp

```diff
@@ -899,6 +1081,65 @@ void IOPlacer::defineSlots()
   }
 }
 
+void IOPlacer::defineSlotsPolygon()
+{
+  for (auto line : core_->getDieAreaEdges()) {
+    bool is_vertical_pin = (line.pt0().getY() == line.pt1().getY());
+    if (is_vertical_pin) {
+      findSlotsPolygon(ver_layers_, line);
+    } else {
+      findSlotsPolygon(hor_layers_, line);
+    }
+  }
```

Comment:
It seems that only this block of code is different when comparing wiwth `defineSlots`. Merge both functions too.